src/Facade/ResourceBuilders/RetailerDemListActivityResourceBuilder.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace Linn.DemStock.Facade.ResourceBuilders
+{
+    using Linn.Common.Facade;
+    using Linn.DemStock.Domain;
+    using Linn.DemStock.Domain.Exceptions;
+    using Linn.DemStock.Domain.RetailerDemListActivities;
+    using Linn.DemStock.Resources;
+
+    public class RetailerDemListActivityResourceBuilder : IResourceBuilder<Activity>
+    {
+        public RetailerDemListActivityResource Build(Activity activity)
+        {
+            switch (activity)
+            {
+                case UpdateRootProductActivity updateRootProductActivity:
+                    return new UpdateRootProductActivityResource
```

Comment:
This is an ideal candidate for @mr-sandy's favourite pattern, The Visitor Pattern!

Reply:
Pretty sure visitor would also remove the need for `InvalidActivityException`